CMakeLists.txt

```diff
@@ -59,7 +59,8 @@ cmake_minimum_required(VERSION 2.6.4)
 # that the subdirectories will at least use the "C" language, so
 # indicate that now.  Doing this early initializes the CFLAGS
 # so they won't change later.
-project(CFETOP C)
+# Note: this line defines the CFE_SOURCE_DIR variable.
+project(CFE C)
```

Comment:
I really like your suggestion of using `project` to initialize the `CFE_SOURCE_DIR` variable. I have also added a note to help developers to find this line in case they want to know where it comes from. Hope this makes sense.